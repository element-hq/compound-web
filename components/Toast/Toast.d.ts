import { default as React } from '../../../node_modules/react';
export declare const Toast: React.ForwardRefExoticComponent<{
    className?: string;
} & React.HTMLAttributes<HTMLDivElement> & React.RefAttributes<HTMLDivElement>>;
//# sourceMappingURL=Toast.d.ts.map