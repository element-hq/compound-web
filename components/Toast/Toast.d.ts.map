{"version":3,"file":"Toast.d.ts","sourceRoot":"","sources":["../../../src/components/Toast/Toast.tsx"],"names":[],"mappings":"AAQA,OAAO,KAAwC,MAAM,OAAO,CAAC;AAO7D,eAAO,MAAM,KAAK;gBAHJ,MAAM;+EAalB,CAAC"}