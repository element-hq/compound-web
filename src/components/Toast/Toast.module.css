/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

.toast-container {
  inline-size: fit-content;
  background-color: var(--cpd-color-alpha-gray-1300);
  color: var(--cpd-color-text-on-solid-primary);
  border-radius: 99px;
  font-size: var(--cpd-font-body-sm-medium);
  padding: var(--cpd-space-2x) var(--cpd-space-4x);
}
