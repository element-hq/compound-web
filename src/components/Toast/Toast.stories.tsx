/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { type Meta, type StoryObj } from "@storybook/react-vite";

import { Toast as ToastComponent } from "./Toast";

const meta = {
  title: "Toast",
  component: ToastComponent,
  tags: ["autodocs"],
  argTypes: {},
  args: {
    children: "Would you like some toast?",
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/design/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?node-id=3627-42633&t=m0RMwUTXkukgU29g-0",
    },
  },
} satisfies Meta<typeof ToastComponent>;
export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
  parameters: {},
};
