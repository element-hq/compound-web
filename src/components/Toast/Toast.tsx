/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import classnames from "classnames";
import React, { forwardRef, PropsWithChildren } from "react";
import styles from "./Toast.module.css";

type ToastProps = {
  className?: string;
} & React.HTMLAttributes<HTMLDivElement>;

export const Toast = forwardRef<HTMLDivElement, ToastProps>(function Toast(
  { children, className, ...props }: PropsWithChildren<ToastProps>,
  ref,
) {
  const classes = classnames(styles["toast-container"], className);
  return (
    <div {...props} className={classes} ref={ref}>
      {children}
    </div>
  );
});
