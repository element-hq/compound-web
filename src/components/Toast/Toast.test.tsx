/*
Copyright 2024 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";

import { Toast } from "./Toast";

describe("Toast", () => {
  it("renders", () => {
    const { asFragment } = render(<Toast />);
    expect(asFragment()).toMatchSnapshot();
  });
});
