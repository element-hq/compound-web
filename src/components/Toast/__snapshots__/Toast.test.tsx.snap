// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast > renders 1`] = `
<DocumentFragment>
  <div
    class="_toast-container_78370f"
  />
</DocumentFragment>
`;
